WordPress/Classes/Services/AccountService.m

```diff
@@ -262,6 +267,20 @@ - (WPAccount *)findAccountWithUserID:(NSNumber *)userID
     return [results firstObject];
 }
 
+- (void)syncAccountDetailsAndCreateAccount:(NSString *)authToken
+                                   success:(void (^)(WPAccount * _Nonnull))success
+                                   failure:(void (^)(NSError * _Nonnull))failure
+{
+    WordPressComRestApi *api = [[WordPressComRestApi alloc] initWithOAuthToken:authToken userAgent:[WPUserAgent defaultUserAgent]];
```

Comment:
Just creating a new `WordPressComRestApi` object directly means it can't be mocked (it will always hit `public-api.wordpress.com`). It should be replaced with this (what we do everywhere else):

```suggestion
    WordPressComRestApi *api = [WordPressComRestApi defaultApiWithOAuthToken:authToken userAgent:[WPUserAgent defaultUserAgent] localeKey:[WordPressComRestApi LocaleKeyDefault]];
```

This allows the mocked base URL to be injected.